### PR TITLE
feat(panel): add '--calcite-panel-content-space' style

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #10119

## Summary

- add `--calcite-panel-content-space` style.
- adds `relative` positioning to content so that absolutely positioned elements are contained within the panel content. (shell use case)
- add e2e test

This will allow the dialog component to set padding on the content of the internal panel without having to surround the content with a div which messes up the placement of the FAB within panel.

### CODE REVIEW

**Review by alisonailea — CHANGES_REQUESTED** [review]
This token name is not consistent with what is in Figma. I'd like to get @SkyeSeitz and @ashetland's guidance. I think (for now) we should prefer the E2E test for variables over chromatic theming but if you do want to add chromatic tests please use https://github.com/Esri/calcite-design-system/wiki/component-tokens for reference. ~~Also, if you have time could you add the other panel tokens while you're here?~~

~~--calcite-panel-background-color~~
~~--calcite-panel-footer-space~~
~~--calcite-panel-text-color~~

**Comment by jcfranco** [comment]
> Also, if you have time could you add the other panel tokens while you're here?
> 
> --calcite-panel-background-color
> --calcite-panel-footer-space
> --calcite-panel-text-color

@alisonailea To keep the PR focused and avoid scope creep, let's handle these changes separately.

**Comment by driskull** [comment]
> This token name is not consistent with what is in Figma. 

How is it not consistent? `calcite-dialog-content-space` was just added and this is nearly identical. Can you elaborate here @alisonailea?

> I think (for now) we should prefer the E2E test for variables over chromatic theming

The e2e test only checks that the property is being applied. It doesn't actually verify things are looking as they are supposed to. Especially when panel is used inside dialog and other components.

File: packages/calcite-components/src/components/panel/panel.stories.ts [hunk 1]
```diff
@@ -253,6 +253,22 @@ export const footerPadding_TestOnly = (): string =>
     </calcite-panel>
   </div>`;
 
+export const contentSpace = (): string =>
```
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/panel/panel.stories.ts:256]: Why would we avoid testing that this is working visually?

**Comment by alisonailea** [comment]
It looks like Aaron and Skye updated the Figma UI Kit to align Figma with the `-content-space` variable after I talked to them yesterday, so that's no longer an issue. But when adding tokens it's important to always double check the Figma UI Kit file and talk with design if there are inconsistencies with code.

All we actually care about in the case of variables is that the expected property changes when the variable does, so E2E tests work pretty well. Doing a visual check as well is nice but the exact visual layout could change and the variable would still be valid. Additionally, there are concerns over adding additional Chromatic tests right now which is why we are moving all chromatic tests to a single storybook file `custom-theme`. I just merged the PR so it should be available to use now. Follow the `Card.ts` file for a good example.

File: packages/calcite-components/src/components/panel/panel.stories.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/panel/panel.stories.ts:256]: I'd like to hold off on moving this to a combined story theme for now if possible. There are other theme stories in here as well.

**Review by alisonailea — APPROVED** [review]
🎉

**Pull request merged**